app/controlplane/internal/authz/authz_test.go

```diff
@@ -0,0 +1,163 @@
+			},
+			wantErr: true,
+		},
+		{
+			name: "happy path",
+			subject: &SubjectAPIToken{
+				ID: uuid.NewString(),
+			},
+			policies: []*Policy{
+				PolicyWorkflowContractList,
+				PolicyReferrerRead,
+			},
+		},
+		{
+			name: "another happy path",
+			subject: &SubjectAPIToken{
+				ID: uuid.NewString(),
+			},
+			policies: []*Policy{
+				PolicyWorkflowContractList,
+				PolicyWorkflowContractRead,
+				PolicyWorkflowContractUpdate,
+				PolicyArtifactDownload,
+			},
+		},
+	}
+
+	for _, tc := range testcases {
+		t.Run(tc.name, func(t *testing.T) {
+			enforcer, closer := testEnforcer(t)
+			defer closer.Close()
+
+			err := enforcer.AddPolicies(tc.subject, tc.policies...)
+			if tc.wantErr {
+				assert.Error(t, err)
+				return
+			}
+
+			require.NoError(t, err)
+
+			for _, p := range tc.policies {
+				ok := enforcer.HasPolicy(tc.subject.String(), p.Resource, p.Action)
+				assert.True(t, ok, fmt.Sprintf("policy %s:%s not found", p.Resource, p.Action))
+			}
+		})
+	}
+}
+
+func TestAddPoliciesDuplication(t *testing.T) {
+	want := []*Policy{
+		PolicyWorkflowContractList,
+		PolicyWorkflowContractRead,
+	}
+
+	enforcer, closer := testEnforcer(t)
+	defer closer.Close()
+	sub := &SubjectAPIToken{ID: uuid.NewString()}
+
+	err := enforcer.AddPolicies(sub, want...)
+	require.NoError(t, err)
+	got := enforcer.GetFilteredPolicy(0, sub.String())
+	assert.Len(t, got, 2)
+
+	// Update the list of policies we want to add by appending an extra one
+	want = append(want, PolicyWorkflowContractUpdate)
+	// AddPolicies only add the policies that are not already present preventing duplication
+	err = enforcer.AddPolicies(sub, want...)
+	assert.NoError(t, err)
+	got = enforcer.GetFilteredPolicy(0, sub.String())
+	assert.Len(t, got, 3)
+}
+
+func TestClearPolicies(t *testing.T) {
+	want := []*Policy{
+		PolicyWorkflowContractList,
+		PolicyWorkflowContractRead,
+	}
+
+	enforcer, closer := testEnforcer(t)
+	defer closer.Close()
+	sub := &SubjectAPIToken{ID: uuid.NewString()}
+	sub2 := &SubjectAPIToken{ID: uuid.NewString()}
+
+	// Create policies for two different subjects
+	err := enforcer.AddPolicies(sub, want...)
+	require.NoError(t, err)
+	err = enforcer.AddPolicies(sub2, want...)
+	require.NoError(t, err)
+	// Each have 2 items
+	got := enforcer.GetFilteredPolicy(0, sub.String())
+	assert.Len(t, got, 2)
+	// 4 in total
+	got = enforcer.GetPolicy()
+	assert.Len(t, got, 4)
+
+	// Clear all the policies for the subject
+	err = enforcer.ClearPolicies(sub)
+	assert.NoError(t, err)
+	// there should be no policies left for this user
+	got = enforcer.GetFilteredPolicy(0, sub.String())
+	assert.Len(t, got, 0)
+	// but the other user should still have 2
+	got = enforcer.GetFilteredPolicy(0, sub2.String())
+	assert.Len(t, got, 2)
+}
+
+func testEnforcer(t *testing.T) (*Enforcer, io.Closer) {
+	policyFilepath := filepath.Join(t.TempDir(), "policy.csv")
+	// create the file if it doesn't exist
+	f, err := os.Create(policyFilepath)
```

Comment:
small note: if the file exists, it will be truncated. If this is intentional, ok. The truncation can happen if you call `testEnforcer` against the same `t` multiple times, as `t.TempDir()` will return the same directory given the same `t`.

Reply:
I'll randomize this. Thanks!